internal/database/sqlcommon/pin_sql.go

```diff
@@ -93,13 +93,13 @@ func (s *SQLCommon) UpsertPin(ctx context.Context, pin *core.Pin) (err error) {
 }
 
 func (s *SQLCommon) attemptPinInsert(ctx context.Context, tx *dbsql.TXWrapper, pin *core.Pin) (err error) {
-	pin.Sequence, err = s.InsertTx(ctx, pinsTable, tx,
+	requestConflictEmptyResult := true
+	pin.Sequence, err = s.InsertTxExt(ctx, pinsTable, tx,
 		s.setPinInsertValues(sq.Insert(pinsTable).Columns(pinColumns...), pin),
 		func() {
 			log.L(ctx).Debugf("Triggering creation event for pin %d", pin.Sequence)
 			s.callbacks.OrderedCollectionNSEvent(database.CollectionPins, core.ChangeEventTypeCreated, pin.Namespace, pin.Sequence)
-		},
-	)
+		}, requestConflictEmptyResult)
```

Comment:
Do we log anything when the insert is a no-op? Seems good to at least be aware for debugging purposes.